examples/basic/main.go

```diff
@@ -0,0 +1,170 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/momentohq/client-sdk-go/auth"
+	"github.com/momentohq/client-sdk-go/config"
+	"github.com/momentohq/client-sdk-go/momento"
+	momento_redis "github.com/momentohq/momento-go-redis-client/momento-redis"
+	"github.com/redis/go-redis/v9"
+)
+
+var useRedis bool
+
+type RedisOptions struct {
+	host string
+	port string
+}
+
+type MomentoOptions struct {
+	cacheName         string
+	authToken         string
+	defaultTTlSeconds int
+}
+
+var options interface{}
+var momentoCacheClient momento.CacheClient
+var momentoCacheName string
+
+func main() {
+	parseFlags()
+
+	// Based on the -useRedis command line flag, we initialize either the
+	// Redis client or the Momento backed Redis client
+	var client redis.Cmdable
+
+	switch o := options.(type) {
+	case *RedisOptions:
+		client = initGoRedisClient(*o)
+		fmt.Println("\nUsing Redis as a backend for go-redis with host " + o.host + " and port " + o.port)
+
+	case *MomentoOptions:
+		client = initMomentoRedisClient(*o)
+		fmt.Println("\nUsing Momento as a backend for go-redis with cache name \"" + o.cacheName + "\"")
+
+	default:
+		panic("nothing found")
+	}
+	fmt.Println("\n-------------------------------------------------")
+
+	key := "Momento"
+	sResp := client.Set(context.Background(), key, "cache", 60*time.Second)
```

Comment:
Instead of constructing a new context for each method call, we usually define a single `ctx` variable at the top of `main()` to pass around instead. We're not really using context at this point, but if we ever do we'll need it to be a persistent object.